_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -1034,221 +1034,265 @@ public static function update_module( $data ) {
-				$updated = $grouped_options != $grouped_options_current ? AtD_update_setting( $user_id, $atd_option, $ignored_phrases ) : true;
-				break;
+					$user_id = get_current_user_id();
+					$grouped_options_current = AtD_get_options( $user_id, $atd_option );
+					unset( $grouped_options_current['name'] );
+					$grouped_options = $grouped_options_current;
+					if ( $value && ! isset( $grouped_options [ $option ] ) ) {
+						$grouped_options [ $option ] = $value;
+					} elseif ( ! $value && isset( $grouped_options [ $option ] ) ) {
+						unset( $grouped_options [ $option ] );
+					}
+					// If option value was the same, consider it done, otherwise try to update it.
+					$options_to_save = implode( ',', array_keys( $grouped_options ) );
+					$updated = $grouped_options != $grouped_options_current ? AtD_update_setting( $user_id, $atd_option, $options_to_save ) : true;
+					break;
 
-			default:
-				// If option value was the same, consider it done.
-				$updated = get_option( $option ) != $value ? update_option( $option, $value ) : true;
-				break;
+				case 'ignored_phrases':
+				case 'unignore_phrase':
+					$user_id = get_current_user_id();
+					$atd_option = 'AtD_ignored_phrases';
+					$grouped_options = $grouped_options_current = explode( ',', AtD_get_setting( $user_id, $atd_option ) );
+					if ( 'ignored_phrases' == $option ) {
+						$grouped_options[] = $value;
+					} else {
+						$index = array_search( $value, $grouped_options );
+						if ( false !== $index ) {
+							unset( $grouped_options[ $index ] );
+							$grouped_options = array_values( $grouped_options );
+						}
+					}
+					$ignored_phrases = implode( ',', array_filter( array_map( 'strip_tags', $grouped_options ) ) );
+					$updated = $grouped_options != $grouped_options_current ? AtD_update_setting( $user_id, $atd_option, $ignored_phrases ) : true;
+					break;
+
+				default:
+					// If option value was the same, consider it done.
+					$updated = get_option( $option ) != $value ? update_option( $option, $value ) : true;
+					break;
+			}
+
+			// The option was not updated.
+			if ( ! $updated ) {
+				$not_updated[ $option ] = $error;
+			}
 		}
 
-		// The option was not updated.
-		if ( ! $updated ) {
-			return new WP_Error( 'module_option_not_updated', esc_html( $error ), array( 'status' => 400 ) );
+		if ( empty( $invalid ) && empty( $not_updated ) ) {
+			// The option was updated.
+			return rest_ensure_response( $response );
+		} else {
+			$invalid_count = count( $invalid );
+			$not_updated_count = count( $not_updated );
+			$error = '';
+			if ( $invalid_count > 0 ) {
+				$invalid_last = array_pop( $invalid );
+				$invalid_text = $invalid_count > 1 ? sprintf(
+					/* Translators: first variable is a list followed by a last item. Example: dog, cat and bird. */
+					__( '%s and %s', 'jetpack' ),
+					join( ', ', $invalid ), $invalid_last ) : $invalid_last;
```

Comment:
I agree that it could be different. Maybe it could just say "Invalid: op1, op2, op3." or "Not updated: op4, op5, op6" without making the difference between singular and plural.